hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanUploadTask.groovy

```diff
@@ -51,7 +53,28 @@ class AzkabanUploadTask extends DefaultTask {
     if (azkProject.azkabanUrl == null) {
       throw new GradleException("Please set azkaban.url in the .azkabanPlugin.json file in your project directory.");
     }
-    uploadToAzkaban(AzkabanHelper.readSession());
+    authenticateAndUploadToAzkaban(AzkabanHelper.readSession());
+  }
+
+  /**
+   * Authenticates session and uploads the zip file to Azkaban.
+   *
+   * @param sessionId The Azkaban session id. If this is null, an attempt will be made to login to Azkaban.
+   */
+  void authenticateAndUploadToAzkaban(String sessionId) {
+    // If no previous session is available, obtain a session id from server by sending login credentials.
+    if (sessionId == null) {
+      logger.lifecycle("No previous session found. Logging into Azkaban:");
+      logger.lifecycle("Azkaban URL: ${azkProject.azkabanUrl}");
+      logger.lifecycle("Azkaban user name: ${azkProject.azkabanUserName}");
+      logger.lifecycle("Azkaban project name: ${azkProject.azkabanProjName}");
+      logger.lifecycle("Azkaban password: ");
+      sessionId = AzkabanHelper.azkabanLogin(azkProject.azkabanUrl, azkProject.azkabanUserName, System.console().readPassword(" Input>"));
+    }
```

Comment:
Ok, cool. I tried a bunch of things in order to avoid the build message peeping in while the tasks are running. There is currently no support in Gradle to avoid it.